cmd/dockerregistry/main.go

```diff
@@ -16,6 +18,18 @@ import (
 )
 
 func main() {
+	// Start the profiling server, if requested.
+	profilingAddr := os.Getenv("REGISTRY_PROFILING_ADDR")
```

Comment:
I don't think this will work. Environment variables prefixed with `REGISTRY_` are interpreted specially by docker. If set, the registry will terminate with an error such as `unsupported variable`. You either need to change to prefix or unset the env var right after being processed.